frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/RediscoverSuggester.java

```diff
@@ -56,31 +51,25 @@ public void performIfNotDeclined() {
     }
   }
 
+  @UIEffect
   private void queueSuggestion(Path macheteFilePath) {
     var yesNo = MessageDialogBuilder.YesNo.yesNo(
         getString("string.GitMachete.RediscoverSuggester.dialog.title"),
         getString("string.GitMachete.RediscoverSuggester.dialog.question"));
 
-    new Task.Backgroundable(project, getString("string.GitMachete.RediscoverSuggester.task-title")) {
-      @Override
-      public void run(ProgressIndicator indicator) {
-        GuiUtils.invokeLaterIfNeeded(() -> {
-          switch (yesNo.show()) {
-            case Messages.YES :
-              LOG.info("Enqueueing rediscover");
-              discoverOperation.run();
-              break;
-            case Messages.NO : // closing dialog goes here too
-              LOG.info("Rediscover declined from dialog");
-              setFileModificationDate(macheteFilePath, System.currentTimeMillis());
-              break;
-            default :
-              LOG.info("Unknown response message");
-              break;
-          }
-        }, ModalityState.NON_MODAL);
-      }
-    }.queue();
+    switch (yesNo.show()) {
+      case Messages.YES :
+        LOG.info("Enqueueing rediscover");
+        discoverOperation.run();
```

Comment:
Hmmm I'd rename it to `queueDiscoverOperation` to signalize that it's going to have an asynchronous part; then the entire method can remain `queueSuggestion`